appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecService.java

```diff
@@ -231,9 +231,10 @@ public Registration addScanEventListener(AppSecScanEventListener listener) {
     public CompletableFuture<Void> scanForVulnerabilities() {
         checkForInitialization();
         Executor executor = configuration.getTaskExecutor();
+        boolean isFlow = isFlow();
```

Comment:
Something doesn't feel right about passing this boolean all throughout the method chain. Can't we get that information inside `GitHubService`? Or even do we need this information? We can keep a cache of both, in case we want to add support for MPR we will need both anyway.

Reply:
Yes, passing through the boolean all throughout the method chain is not the best solution. We can get this value with `AppSecService.getInstance().isFlow()` is `GitHubService` which can simplify the solution.
I would keep this check because in other case, if we keep cache of both, we would call the GitHub API twice and store data what is unnecessary when not using MPR.

Reply:
Sounds reasonable. Since it's only used in a single method, let's get it there from the singleton 👍

Reply:
Done.